anesthetic/plot.py

```diff
@@ -388,6 +390,9 @@ def contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
     ax.contour(x[i], y[j], pdf[numpy.ix_(j, i)], contours,
                vmin=0, vmax=1.2, linewidths=0.5, colors='k', zorder=zorder+2,
                *args, **kwargs)
+    ax.patches += [plt.Rectangle((0, 0), 1, 1, fc=cmap(0.999), ec=cmap(0.33),
```

Comment:
What's the reason for the `+1` and `+2` to `zorder` in the contour plots? 

The way they are determined from the axes children they behave strangely when there is also a legend or another plot in that axis.

Reply:
Originally I think it was to make sure that the respective colours were on top (particularly with regards to samples), but it might be more consistent let them be handled automatically, and give the user charge of ``zorder`` vias the ``*args, **kwargs``. Does removing them give any immediately odd behaviour?

Reply:
Ok, the odd thing is that the contour lines now always show, even with multiple contours on the same axis. This is because the default `zorder` for `conourf` is 1 and for `contour` is 2. 

I think kwarg popping zorder and setting the default to 1 is the better thing to do here, though.

Reply:
Ah, I knew there was a reason I did this. For our case, it is definitely correct to override this default.